src/Bundle/Context/Initiator/LegacyRequestContextInitiator.php

```diff
@@ -27,6 +28,7 @@ public function __construct(
         private RegistryInterface $resourceRegistry,
         private RequestConfigurationFactoryInterface $requestConfigurationFactory,
         private RequestContextInitiatorInterface $decorated,
+        private ?VarsResolverInterface $varsResolver = null,
```

Comment:
Does this place need a deprecation notice, that not providing VarsResolver is not supported anymore and will be required at some future version?

Reply:
yeah, I think so. I'll add that on the two ones.